lib/hiptest-publisher/options_parser.rb

```diff
@@ -132,10 +136,15 @@ def command_line_used(exclude: [])
       "--#{key.to_s.gsub('_', '-')}=#{self[key]}"
     end.compact
 
-    return "hiptest-publisher #{args.join(' ')}".strip
+    "hiptest-publisher #{args.join(' ')}".strip
+  end
+
+  def uids?
+    __cli_args.include?(:uids) || __config_args.include?(:uids)
   end
 
   def normalize!(reporter = nil)
+    self.uids = true if test_run? && !uids?
```

Comment:
That's a bit confusing (`uids = true if !uids?` when reading quickly).

Maybe renaming `uids?` as `forced_uids?`